lib/src/functions/map.dart

```diff
@@ -53,14 +53,39 @@ final _set = BuiltInCallable.overloadedFunction("set", {
     } else if (args.length == 1) {
       throw SassScriptException("Expected \$args to contain a value.");
     }
-    return _setImpl(map, args.sublist(0, args.length - 1), args.last);
+    return _modify(
+        map, args.sublist(0, args.length - 1), (oldValue) => args.last);
   },
 });
 
-final _merge = _function("merge", r"$map1, $map2", (arguments) {
-  var map1 = arguments[0].assertMap("map1");
-  var map2 = arguments[1].assertMap("map2");
-  return SassMap({...map1.contents, ...map2.contents});
+final _merge = BuiltInCallable.overloadedFunction("merge", {
+  r"$map1, $map2": (arguments) {
+    var map1 = arguments[0].assertMap("map1");
+    var map2 = arguments[1].assertMap("map2");
+    return SassMap({...map1.contents, ...map2.contents});
+  },
+  r"$map1, $args...": (arguments) {
+    var map1 = arguments[0].assertMap("map1");
+    var args = arguments[1].asList;
+    if (args.isEmpty) {
+      throw SassScriptException("Expected \$args to contain a key.");
+    } else if (args.length == 1) {
+      throw SassScriptException("Expected \$args to contain a map.");
+    }
+    var keys = args.sublist(0, args.length - 1);
+    var map2 = args.last.assertMap("map2");
+    SassMap _mergeCallback(Value oldMap) {
+      // If oldMap is null, this indicates there is no nested map at the key.
+      // If this is the case, set the value at the key to map2, which is akin to
+      // merging map2 into an empty map. This behavior is consistent with that
+      // of map.set().
+      return (oldMap == null)
+          ? map2
+          : SassMap({...oldMap.assertMap().contents, ...map2.contents});
```

Comment:
According to the spec, the new map is expected to replace the old value if it's not a map, rather than throwing an error.

Reply:
Ah, good catch.